### PR TITLE
*: move evaluator's helper to expression package.

### PR DESCRIPTION
**Pull request by hanfei1991**
prepare to say good bye to old Eval function.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by hanfei1991** [comment]
@tiancaiamao @zimulala PTAL

**Comment by hanfei1991** [comment]
@tiancaiamao @zimulala PTAL

File: expression/helper_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		Ret  interface{}
+	}{
+		{"2012-12-12 00:00:00", "2012-12-12 00:00:00"},
+		{CurrentTimestamp, time.Unix(1234, 0).Format(types.TimeFormat)},
+		{ZeroTimestamp, "0000-00-00 00:00:00"},
+		{ast.NewValueExpr("2012-12-12 00:00:00"), "2012-12-12 00:00:00"},
+		{ast.NewValueExpr(int64(0)), "0000-00-00 00:00:00"},
+		{ast.NewValueExpr(nil), nil},
+		{&ast.FuncCallExpr{FnName: model.NewCIStr(CurrentTimestamp)}, CurrentTimestamp},
+		//{&ast.UnaryOperationExpr{Op: opcode.Minus, V: ast.NewValueExpr(int64(0))}, "0000-00-00 00:00:00"},
```
Comment (zimulala) [inline expression/helper_test.go]: Why remove this test?

File: expression/helper_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	errTbl := []struct {
+		Expr interface{}
+	}{
+		{"2012-13-12 00:00:00"},
+		{ast.NewValueExpr("2012-13-12 00:00:00")},
+		{ast.NewValueExpr(int64(1))},
+		{&ast.FuncCallExpr{FnName: model.NewCIStr("xxx")}},
+		//{&ast.UnaryOperationExpr{Op: opcode.Minus, V: ast.NewValueExpr(int64(1))}},
```
Comment (zimulala) [inline expression/helper_test.go]: Why remove this test?

File: expression/helper_test.go (hunk 1 above)
Comment (hanfei1991) [inline expression/helper_test.go]: We cannot call EvalAstExpr in test.

Comment (hanfei1991) [inline expression/helper_test.go]: I will try implement a Eval function in expression without Subquery in expression package in future.

Comment (zimulala) [inline expression/helper_test.go]: I don't understand. The `GetTimeValue` function calls `EvalAstExpr`, and the test calls `GetTimeValue`. Why do you say we can't call `EvalAstExpr` in test？

Comment (hanfei1991) [inline expression/helper_test.go]: EvalAstExpr is assigned in plan init function, which cannot be called in expression test.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**